.github/workflows/build-container.yml

```diff
@@ -0,0 +1,74 @@
+name: Build and Push Docker image
+
+on:
+  push:
+    branches: [ master ]
+  repository_dispatch:
+    types:
+      - dispatch-build
+  workflow_dispatch:
+      
+jobs:
+  make-date-tag:
+    runs-on: ubuntu-latest
+    if: startsWith(github.repository, 'opensciencegrid/')
+    outputs:
+      dtag: ${{ steps.mkdatetag.outputs.dtag }}
+    steps:
+    - name: make date tag
+      id: mkdatetag
+      run: echo "::set-output name=dtag::$(date +%Y%m%d-%H%M)"
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [make-date-tag]
+    if: startsWith(github.repository, 'opensciencegrid/')
+    strategy:
+      fail-fast: False
+      matrix:
+        repo: ['latest']
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Generate tag list
+      id: generate-tag-list
+      env:
+        REPO: ${{ matrix.repo }}
+        TIMESTAMP: ${{ needs.make-date-tag.outputs.dtag }}
+      run: |
+        #docker_repo=${GITHUB_REPOSITORY/opensciencegrid\/docker-/opensciencegrid/}
+        docker_repo=opensciencegrid/comanage-utils
+        tag_list=()
+        for registry in hub.opensciencegrid.org docker.io; do
+          for image_tag in "$REPO" "$REPO-$TIMESTAMP"; do
+            tag_list+=("$registry/$docker_repo":"$image_tag")
+          done
+        done
+        # This causes the tag_list array to be comma-separated below,
+        # which is required for build-push-action
+        IFS=,
+        echo "::set-output name=taglist::${tag_list[*]}"
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Log in to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        
+    - name: Log in to OSG Harbor
+      uses: docker/login-action@v1
+      with:
+        registry: hub.opensciencegrid.org
+        username: ${{ secrets.OSG_HARBOR_ROBOT_USER }}
+        password: ${{ secrets.OSG_HARBOR_ROBOT_PASSWORD }}
+
+    - name: Build and push Docker images
+      uses: docker/build-push-action@v2.2.0
+      with:
+        context: .
+        push: true
+        tags: "${{ steps.generate-tag-list.outputs.taglist }}"
```

Comment:
One thing that I've started doing is turning this into two steps, one for Harbor and one for Docker Hub so if Harbor's down, it doesn't blow up the whole push. You can make sure a step always runs, regardless of previous step failures with something like https://github.com/opensciencegrid/images/blob/main/.github/workflows/build-containers.yml#L168

You'll also need to fixup the taglist to separate the hub.opensciencegrid.org vs docker.io tags

Reply:
>two steps, one for Harbor and one for Docker Hub so if Harbor's down, it doesn't blow up the whole push

@brianhlin, would we perhaps get that same behavior you are describing by just putting `docker.io` first in the registry list, before harbor?  (Assuming that Docker Hub being down is a sufficiently rare case that we don't have to worry about.)

I have gone ahead and done that in e315e9b.

I did have a look at your images workflow and i am not feeling really confident about implementing it correctly on the first try.

Given that this PR is holding up SOFTWARE-5154 (the image is not built and thus not available from harbor), i am thinking just to get this PR merged.  Then, if you still want your robustness improvement, and you think that putting `docker.io` first in the list won't do just that, then I can put that in a separate PR as not to hold up the show for SOFTWARE-5154.

What do you think?

Reply:
Sounds reasonable to me!